Shopipy.ServiceManagement/Controllers/ServiceController.cs

```diff
@@ -0,0 +1,104 @@
+using System.Collections;
+using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using Shopipy.CategoryManagement.Services;
+using Shopipy.Persistence.Models;
+using Shopipy.ServiceManagement.DTOs;
+using Shopipy.ServiceManagement.Mappings;
+using Shopipy.ServiceManagement.Services;
+
+namespace Shopipy.ServiceManagement.Controllers;
+
+[ApiController]
+[Route("[controller]")]
+public class ServiceController : ControllerBase
+{
+    private readonly ServiceManagementService _serviceManagementService;
+    private readonly CategoryService _categoryService;
+    private readonly IMapper _mapper;
+
+    public ServiceController(ServiceManagementService serviceManagementService, CategoryService categoryService, IMapper mapper)
+    {
+     _serviceManagementService = serviceManagementService;
+    _categoryService = categoryService;
+    _mapper = mapper;
+    }
+    [HttpGet("{businessId}/services")]
+    public async Task<IActionResult> GetServices(int businessId)
+    {
+        var services = await _serviceManagementService.GetAllServicesInBusiness(businessId);
+        var responseDtos = _mapper.Map<IEnumerable<ServiceResponseDto>>(services);
+    
+        return Ok(responseDtos);
+    }
+
+    [HttpGet("{businessId}/services/{id}")]
+    public async Task<IActionResult> GetService(int businessId, int id)
+    {
+        var service = await _serviceManagementService.GetServiceByIdInBusiness(businessId, id);
+        if (service == null) return NotFound();
+        
+        var responseDto = _mapper.Map<ServiceResponseDto>(service);
+
+        return Ok(responseDto);
+    }
+
+    [HttpGet("{businessId}/services/category/{categoryId}")]
+    public async Task<IActionResult> GetServicesByCategory(int businessId, int categoryId)
+    {
+        var category = await _categoryService.GetCategoryByIdAsync(categoryId);
+        if (category == null)
+        {
+            return NotFound("Category does not exist");
+        }
```

Comment:
Such checking is redundant, it is enough to simply do Linq query with where clause, who cares if user asks for non existant category

Reply:
the app is definitely not gonna crash